### PR TITLE
move clickAndWaitForSelector to utils package

### LINKED ISSUE #29178: Move clickAndWaitForSelector() to the utils
We should move the `clickAndWaitForSelector()` function that was added in https://github.com/woocommerce/woocommerce/pull/29079/files into the utils so it can be consumed by other e2e tests.

### PR DESCRIPTION
**Pull request by rrennick**
### Changes proposed in this Pull Request:

This PR moves the `clickAndWaitForSelector()` function from the order refunds test to the utilities package. To provide more flexibility it adds the timeout length as a third parameter.

The PR also updates code spacing in `page-utils.js` so the spacing is consistent throughout the file.

Closes #29178 .

### How to test the changes in this Pull Request:

1. Run the following
```
cd plugins/woocommerce
pnpm install
pnpx wc-e2e docker:up
pnpx wc-e2e test:e2e
```
2. Tests should pass

### CODE REVIEW

**Comment by rrennick** [comment]
Note: WP Nightly test is failing due to https://core.trac.wordpress.org/changeset/52032.

File: packages/js/e2e-utils/src/page-utils.js [hunk 1]
```diff
@@ -296,3 +297,24 @@ export const selectOrderAction = async ( action ) => {
...
+ * @param {number} timeout Timeout length in milliseconds. Default 5000.
+ * @returns {Promise<void>}
+ */
+export const clickAndWaitForSelector = async ( buttonSelector, resultSelector, timeout = 5000 ) => {
+	await evalAndClick( buttonSelector );
+	await waitForSelector(
+		page,
+		resultSelector,
+		{
+			timeout
```
Comment (zhongruige) [inline packages/js/e2e-utils/src/page-utils.js:317]: I noticed this dropped the `timeout:` named parameter, so wanted to check if that is needed or not here, or if this needs to be something like:

```suggestion
			timeout: timeout
```

**Review by zhongruige — COMMENTED** [review]
Thanks for making these changes @rrennick! Just had a quick question on one line no longer having the `timeout:` field.

File: packages/js/e2e-utils/src/page-utils.js (hunk 1 above)
Comment (rrennick) [reply packages/js/e2e-utils/src/page-utils.js:317]: The code in the e2e-utils/src folder is ES6 which supports [Shorthand property names](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Operators/Object_initializer#new_notations_in_ecmascript_2015). The original is the shorthand of your suggestion.

Comment (rrennick) [reply packages/js/e2e-utils/src/page-utils.js:317]: I felt I should confirm this so I added a bit of test code

```
 export const clickAndWaitForSelector = async ( buttonSelector, resultSelector, timeout = 5000 ) => {
+       console.log( { timeout } );
        await evalAndClick( buttonSelector );
....
 console.log
    { timeout: 5000 }
```

Comment (zhongruige) [reply packages/js/e2e-utils/src/page-utils.js:317]: Thanks for the followup and taking the time to confirm this!

**Comment by zhongruige** [comment]
@rrennick Looks like there's a merge conflict on the `pnpm-lock.yaml` file we'll need to resolve before merging.

**Review by zhongruige — APPROVED** [review]
Changes look great, thanks @rrennick!

**Pull request merged**